source/containers.tex

```diff
@@ -20957,3 +21022,628 @@
+Specializations of \tcode{submdspan_mapping_result}
+are returned by overloads of \tcode{submdspan_mapping}.
+
+\indexlibraryglobal{submdspan_mapping_result}%
+\begin{itemdecl}
+template<class LayoutMapping>
+struct submdspan_mapping_result {
+  [[no_unique_address]] LayoutMapping mapping = LayoutMapping();
+  size_t offset{};
+};
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\tcode{submdspan_mapping_result} has
+the data members and special members specified above.
+It has no base classes or members other than those specified.
+
+\pnum
+\tcode{LayoutMapping} shall meet
+the layout mapping requirements\iref{mdspan.layout.policy.reqmts}.
+\end{itemdescr}
+
+\rSec4[mdspan.submdspan.helpers]{Exposition-only helpers}
+
+\indexlibraryglobal{\exposid{de-ice}}%
+\indexlibraryglobal{\exposid{first_}}%
+\begin{itemdecl}
+template<class T>
+  constexpr T @\exposid{de-ice}@(T val) { return val; }
+template<@\exposconcept{integral-constant-like}@ T>
+  constexpr auto @\exposid{de-ice}@(T) { return T::value; }
+
+template<class IndexType, size_t k, class ... SliceSpecifiers>
+  constexpr IndexType @\exposid{first_}@(SliceSpecifiers... slices);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\mandates
+\tcode{IndexType} is a signed or unsigned integer type.
+
+%FIXME: What is $k$ in this itemdescr? Did we mean the \tcode{k} passed to first_? If so, what do we do for de-ice?
+\pnum
+Let $\phi{}_k$ denote the following value:
+\begin{itemize}
+\item
+$s_k$ if $S_k$ models \tcode{\libconcept{convertible_to}<IndexType>}; otherwise
+\item
+\tcode{get<0>($s_k$)}
+if $S_k$ models \tcode{\exposconcept{index-pair-like}<IndexType>}; otherwise
+\item
+\tcode{\exposid{de-ice}($s_k$.offset)}
+if $S_k$ is a specialization of \tcode{strided_slice}; otherwise
+\item
+\tcode{0}.
+\end{itemize}
+
+\pnum
+\expects
+$\phi{}_k$ is representable as a value of type \tcode{IndexType}.
+
+\pnum
+\returns
+\tcode{extents<IndexType>::\exposid{index-cast}($\phi{}_k$)}.
+\end{itemdescr}
+
+\indexlibraryglobal{\exposid{last_}}%
+\begin{itemdecl}
+template<size_t k, class Extents, class ... SliceSpecifiers>
+  constexpr auto @\exposid{last_}@(const Extents& src, SliceSpecifiers... slices);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\mandates
+\tcode{Extents} is a specialization of \tcode{extents}.
+
+\pnum
+Let \tcode{index_type} name the type \tcode{typename Extents::index_type}.
+
+%FIXME: What is $k$ in this itemdescr? Don't we want \tcode{k} and not $k$?
+\pnum
+Let $\lambda{}_k$ denote the following value:
+\begin{itemize}
+\item
+\tcode{\exposid{de-ice}($s_k$) + 1}
+if $S_k$ models \tcode{\libconcept{convertible_to}<index_type>}; otherwise
+\item
+\tcode{get<1>($s_k$)}
+if $S_k$ models \tcode{\exposconcept{index-pair-like}<index_type>}; otherwise
+\item
+\tcode{\exposid{de-ice}($s_k$.offset)} \tcode{+}
+\tcode{\exposid{de-ice}($s_k$.extent)}
+if $S_k$ is a specialization of \tcode{strided_slice}; otherwise
+\item
+%FIXME: Note that the paper uses \tcode{k} here.
+\tcode{src.extent(k)}.
```

Comment:
Yes, because In the _`src-indices`_ case, there is no parameter `k` to confuse it with; it's clear that we mean the ranging index /_k_/ introduced in the /_Returns_/ clause p11 (or the 1st bullet p11.1, but that's a different problem).